invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/InvitationResultItem.js

```diff
@@ -6,146 +6,84 @@
  * under the terms of the MIT License; see LICENSE file for more details.
  */
 
+import { CommunityInvitationsApi } from "@js/invenio_communities/api/InvitationsApi";
+import ActionDropdown from "../components/ActionDropdown";
 import React, { Component } from "react";
 import { DateTime } from "luxon";
-import {
-  Table,
-  Grid,
-  Container,
-  List,
-  Icon,
-  Loader, Dropdown,
-} from 'semantic-ui-react';
+import { Table, Container, Header } from "semantic-ui-react";
 import { Image } from "react-invenio-forms";
-import { ActionButtons } from "./InvitationActionButtons";
+import { ActionButtons } from "./request_actions/InvitationActionButtons";
 import PropTypes from "prop-types";
-import { ErrorPopup } from "../components/ErrorPopup";
 import { i18next } from "@translations/invenio_communities/i18next";
-import { SuccessIcon } from "../components/SuccessIcon";
-import _sample from "lodash/sample";
+import {
+  RequestActionController,
+} from "@js/invenio_requests";
 
-const dropdownOptionsGenerator = (value) => {
-  return Object.entries(value).map(([key, settings]) => {
-    return {
-      key: key,
-      text: settings.title,
-      value: key,
-      content: (
-        <List>
-          <List.Item>
-            <List.Content>
-              <List.Header>{settings.title}</List.Header>
-              <List.Description>{settings.description}</List.Description>
-            </List.Content>
-          </List.Item>
-        </List>
-      ),
-    };
-  });
-};
 
 const formattedTime = (expires_at) =>
   DateTime.fromISO(expires_at).setLocale(i18next.language).toRelative();
 
-export const InvitationResultItem = ({
-  invitation,
-  onRoleChange,
-  isLoading,
-  error,
-  success,
-  onSuccessTimeOut,
-  onErrorClose,
-  roles,
-  onView,
-  onReInvite,
-  onCancel,
-}) => {
-  const { receiver, status, expires_at, links } = invitation;
+export class InvitationResultItem extends Component {
+  constructor(props) {
+    super(props);
+    const { result } = this.props;
+    this.state = { invitation: result };
+    this.invitationsApi = new CommunityInvitationsApi(result.links);
```

Comment:
for reviewers: do you have any information if there will be any separate resource to update invitation role or the generic request API will be used?

Reply:
We are missing a new backend endpoint for the invitations